src/main/java/com/amihaiemil/eoyaml/FirstCommentFound.java

```diff
@@ -146,12 +146,10 @@ public YamlNode toYamlNode(
                         if(line.trimmed().startsWith("#")) {
                             comment.add(line);
                         }
-                    } else {
+                    } else if (!line.trimmed().isEmpty()){
```

Comment:
@amihaiemil Here needed to be changed, because in case of a yaml stream it will prevent to search for the first comment in the above document(s). Now it starts above the '---' and go backwards until there are no more lines starting with '#'. (empty lines are ignored).